fermenter-mda/src/main/java/org/technologybrewery/fermenter/mda/GenerateSourcesMojo.java

```diff
@@ -630,4 +633,10 @@ protected File getMainSourceRoot() {
     protected File getGeneratedSourceRoot() {
         return generatedSourceRoot;
     }
+
+    public String getRootArtifactId() {
+        MavenProject topLevelProject = session.getTopLevelProject();
```

Comment:
Q: I did some quick googling but can't seem to verify-- Does this work as expected when the current working directory is a submodule, or when invoked from outside the module directory structure (ie: from $HOME/random, execute `mvn clean install -f $HOME/my_project/sub_module`?  Just sanity checking that the function does what I think it does

Reply:
^ the above has been confirmed through invoking from outside the module directory structure & from within a submodule